src/test/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActorTest.scala

```diff
@@ -314,163 +306,46 @@ class TaskLauncherActorTest extends AkkaUnitTest {
+      launcherRef ! InstanceDeleted(f.provisionedInstance, Some(f.scheduledInstance.state), Seq.empty)
 
-      // task status update
-      Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(update.instance))
-      val counts = sendUpdate(launcherRef, update)
-
-      assert(counts.instancesLeftToLaunch == expectedCounts.instancesLeftToLaunch)
-      assert(counts.finalInstanceCount == expectedCounts.finalInstanceCount)
+      Mockito.verify(offerReviver).reviveOffers()
     }
 
-    for (
-      update <- TaskConditionMapping.Gone.toSeq.map(reason => TaskStatusUpdateTestHelper.lost(reason, f.marathonInstance))
-        .union(Seq(
-          TaskStatusUpdateTestHelper.finished(f.marathonInstance),
-          TaskStatusUpdateTestHelper.killed(f.marathonInstance),
-          TaskStatusUpdateTestHelper.error(f.marathonInstance)))
-    ) {
-      s"Terminated task (${update.simpleName} with ${update.reason} is removed" in new Fixture {
-        Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(f.marathonInstance))
-
-        val launcherRef = createLauncherRef()
-        launcherRef ! RateLimiterActor.DelayUpdate(f.app, clock.now())
-
-        // wait for startup
-        (launcherRef ? TaskLauncherActor.GetCount).futureValue.asInstanceOf[QueuedInstanceInfo]
-
-        // task status update
-        val counts = sendUpdate(launcherRef, update.wrapped)
-
-        assert(!counts.inProgress)
-        assert(counts.instancesLeftToLaunch == 0)
-
-        verifyClean()
-      }
-    }
-
-    for (
-      reason <- TaskConditionMapping.Unreachable
-    ) {
-      s"TemporarilyUnreachable task ($reason) is NOT removed" in new Fixture {
-        val update = TaskStatusUpdateTestHelper.lost(reason, f.marathonInstance, timestamp = clock.now())
-        Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(f.marathonInstance))
-
-        val launcherRef = createLauncherRef()
-        launcherRef ! RateLimiterActor.DelayUpdate(f.app, clock.now())
-
-        // wait for startup
-        (launcherRef ? TaskLauncherActor.GetCount).futureValue.asInstanceOf[QueuedInstanceInfo]
-
-        // task status update
-        val counts = sendUpdate(launcherRef, update.wrapped)
-        assert(!counts.inProgress)
-        assert(counts.finalInstanceCount == 1)
-        assert(counts.instancesLeftToLaunch == 0)
-
-        verifyClean()
-      }
-    }
-
-    "Updated task is reflected in counts" in new Fixture {
-      val update = TaskStatusUpdateTestHelper.runningHealthy(f.marathonInstance)
-      val expectedCounts = QueuedInstanceInfo(f.app, inProgress = false, 0, 1, Timestamp(0), Timestamp(0))
-
-      Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(f.marathonInstance))
-
+    "do instance tracker call after InstanceChange event" in new Fixture {
```

Comment:
How about `should { "call instance tracker after InstanceChange event" in {} }`?